### PR TITLE
Remove usage deprecated methods

### PR DESCRIPTION
**Pull request by putrenkov**
**A Brief Overview**
Refactor and remove deprecated method

**Link to QA issue**
BroadleafCommerce/QA#3878

### CODE REVIEW

File: admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/security/handler/AdminUserCustomPersistenceHandler.java [hunk 1]
```diff
@@ -57,9 +58,12 @@
 
     @Resource(name = "blAdminSecurityRemoteService")
     protected SecurityVerifier adminRemoteSecurityService;
+    
+    @Autowired
+    protected Environment environment;
 
     protected boolean getRequireUniqueEmailAddress() {
-        return BLCSystemProperty.resolveBooleanSystemProperty("admin.user.requireUniqueEmailAddress");
+        return  environment.getProperty("admin.user.requireUniqueEmailAddress").isEmpty();
```
Comment (cja769) [inline admin/broadleaf-open-admin-platform/src/main/java/org/broadleafcommerce/openadmin/server/security/handler/AdminUserCustomPersistenceHandler.java]: This should not have a `isEmpty`. That is a boolean property. Should be 
```
return environment.getProperty("admin.user.requireUniqueEmailAddress", Boolean.class);
```

File: core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/controller/account/BroadleafManageWishlistController.java [hunk 2]
```diff
@@ -48,7 +49,7 @@
     protected static String accountWishlistRedirect = "redirect:/account/wishlist";
 
     public String add(HttpServletRequest request, HttpServletResponse response, Model model,
-            AddToCartItem itemRequest, String wishlistName) throws IOException, AddToCartException, PricingException  {
+                      ConfigurableOrderItemRequest itemRequest, String wishlistName) throws IOException, AddToCartException, PricingException  {
```
Comment (cja769) [inline core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/controller/account/BroadleafManageWishlistController.java]: This needs to be OrderItemRequestDTO instead. First of all for compatibility reasons (i.e. clients may be using this method and sending it AddToCartItem which subclasses OrderItemRequestDTO however it doesn't subclass ConfiguratbleOrderItemRequest) and second of all if we only need it to be an OrderItemRequestDTO then there's not reason to require a more specific implementation

File: core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/controller/cart/BroadleafCartController.java [hunk 3]
```diff
@@ -350,7 +350,7 @@ public String remove(HttpServletRequest request, HttpServletResponse response, M
 
     @Deprecated
     public String remove(HttpServletRequest request, HttpServletResponse response, Model model,
-                         AddToCartItem itemRequest) throws IOException, PricingException, RemoveFromCartException {
+                         ConfigurableOrderItemRequest itemRequest) throws IOException, PricingException, RemoveFromCartException {
```
Comment (cja769) [inline core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/controller/cart/BroadleafCartController.java]: This needs to be OrderItemRequestDTO instead. First of all for compatibility reasons (i.e. clients may be using this method and sending it AddToCartItem which subclasses OrderItemRequestDTO however it doesn't subclass ConfiguratbleOrderItemRequest) and second of all if we only need it to be an OrderItemRequestDTO then there's not reason to require a more specific implementation

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java [hunk 4]
```diff
@@ -28,32 +28,32 @@
     @SuppressWarnings("unchecked")
 
     //TOOD: this should probably be refactored to be called "rows" like in other model objects
-    private List<AddToCartItem> addToCartItems =   LazyList.decorate(
-            new ArrayList<AddToCartItem>(),
-            FactoryUtils.instantiateFactory(AddToCartItem.class));
+    private List<ConfigurableOrderItemRequest> addToCartItems =   LazyList.decorate(
+            new ArrayList<ConfigurableOrderItemRequest>(),
+            FactoryUtils.instantiateFactory(ConfigurableOrderItemRequest.class));
```
Comment (cja769) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java]: These should be OrderItemRequestDTO instead. No need to make it more specific when it doesn't need to be

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java [hunk 5]
```diff
@@ -28,32 +28,32 @@
...
 
     private long productId;
     private long categoryId;
 
     public void setProductId(long productId) {
         this.productId = productId;
-        for(AddToCartItem addToCartItem : addToCartItems) {
+        for(ConfigurableOrderItemRequest addToCartItem : addToCartItems) {
```
Comment (cja769) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java]: These should be OrderItemRequestDTO instead. No need to make it more specific when it doesn't need to be

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java [hunk 6]
```diff
@@ -28,32 +28,32 @@
...
             addToCartItem.setProductId(productId);
         }
     }
 
     public void setCategoryId(long categoryId) {
         this.categoryId = categoryId;
-        for(AddToCartItem addToCartItem : addToCartItems) {
+        for(ConfigurableOrderItemRequest addToCartItem : addToCartItems) {
```
Comment (cja769) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java]: These should be OrderItemRequestDTO instead. No need to make it more specific when it doesn't need to be

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java [hunk 7]
```diff
@@ -28,32 +28,32 @@
...
             addToCartItem.setCategoryId(categoryId);
         }
     }
 
-    public List<AddToCartItem> getAddToCartItems() {
+    public List<ConfigurableOrderItemRequest> getAddToCartItems() {
```
Comment (cja769) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java]: These should be OrderItemRequestDTO instead. No need to make it more specific when it doesn't need to be

File: core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java [hunk 8]
```diff
@@ -28,32 +28,32 @@
...
         return addToCartItems;
     }
 
-    public void setAddToCartItem(List<AddToCartItem> addToCartItems) {
+    public void setAddToCartItem(List<ConfigurableOrderItemRequest> addToCartItems) {
```
Comment (cja769) [inline core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/call/AddToCartItems.java]: These should be OrderItemRequestDTO instead. No need to make it more specific when it doesn't need to be

**Review by cja769 — CHANGES_REQUESTED** [review]
I was afraid that these changes may need to be added into 6.2 however it looks like the changes made shouldn't affect clients. With the requested changes it should be good to merge

File: core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/controller/account/AbstractCustomerAddressController.java [hunk 9]
```diff
@@ -87,8 +91,8 @@ protected void initBinder(HttpServletRequest request, ServletRequestDataBinder b
         initBinderService.configAddressInitBinder(binder);
     }
 
-    protected List<State> populateStates() {
-        return stateService.findStates();
+    protected List<CountrySubdivision> populateStates() {
+        return countrySubdivisionService.findSubdivisions();
```
Comment (cja769) [inline core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/controller/account/AbstractCustomerAddressController.java]: Lets also revert this change. It looks like that in our demo we extend this and set a model attribute on it. Also CountrySubdivision != State therefore the front end mappings setup in demo site aren't setup correctly for this change (even in the PR). We'll catch this with the next pass when we do 6.2

**Comment by putrenkov** [comment]
@cja769 I pushed my changes
